### PR TITLE
docs(spec): add RFC for Message Example Object

### PR DESCRIPTION
**Pull request by char0n**
This is formal proposal for introducing new Message Example Object as originally proposed by a strawman in #606. Champion of this RFC proposal is me (@char0n).

### CODE REVIEW

File: spec/asyncapi.md [hunk 1]
```diff
@@ -1234,6 +1235,55 @@ schemaFormat: 'application/vnd.apache.avro+yaml;version=1.9.0'
 contentType: application/json
 ```
 
+#### <a name="messageExampleObject"></a> Message Example Object
+
+Message Example Object represent example of a [Message Object](#messageObject) and MUST contain either **headers** and/or **payload** fields.
```
Comment (fmvilas) [inline spec/asyncapi.md:1240]:
```suggestion
Message Example Object represents an example of a [Message Object](#messageObject) and MUST contain either **headers** and/or **payload** fields. 
```

File: spec/asyncapi.md [hunk 2]
```diff
@@ -1234,6 +1235,55 @@ schemaFormat: 'application/vnd.apache.avro+yaml;version=1.9.0'
...
 
+#### <a name="messageExampleObject"></a> Message Example Object
+
+Message Example Object represent example of a [Message Object](#messageObject) and MUST contain either **headers** and/or **payload** fields. 
+
+##### Fixed Fields
+
+Field Name | Type | Description
+---|:---:|---
+<a name="messageExampleObjectHeaders"></a>headers | `Map[string, any]` | The value of this field MUST validate against the [Message Object](#messageObject) [headers](#messageObjectHeaders) field.
```
Comment (fmvilas) [inline spec/asyncapi.md:1246]:
```suggestion
<a name="messageExampleObjectHeaders"></a>headers | `Map[string, any]` | The value of this field MUST validate against the [Message Object's headers](#messageObjectHeaders) field. 
```

File: spec/asyncapi.md [hunk 3]
```diff
@@ -1234,6 +1235,55 @@ schemaFormat: 'application/vnd.apache.avro+yaml;version=1.9.0'
...
+#### <a name="messageExampleObject"></a> Message Example Object
+
+Message Example Object represent example of a [Message Object](#messageObject) and MUST contain either **headers** and/or **payload** fields. 
+
+##### Fixed Fields
+
+Field Name | Type | Description
+---|:---:|---
+<a name="messageExampleObjectHeaders"></a>headers | `Map[string, any]` | The value of this field MUST validate against the [Message Object](#messageObject) [headers](#messageObjectHeaders) field. 
+<a name="messageExampleObjectPayload"></a>payload | `any` | The value of this field MUST validate against the [Message Object](#messageObject) [payload](#messageObjectPayload) field.
```
Comment (fmvilas) [inline spec/asyncapi.md:1247]:
```suggestion
<a name="messageExampleObjectPayload"></a>payload | `any` | The value of this field MUST validate against the [Message Object's payload](#messageObjectPayload) field.
```

**Review by fmvilas — CHANGES_REQUESTED** [review]

File: spec/asyncapi.md (hunk 2 above)
Comment (char0n) [reply spec/asyncapi.md:1246]: Sure, look better and `Message Object` string is still searchable by native browser capabilities (CTRL+F)

Comment (fmvilas) [reply spec/asyncapi.md]: Actually, the link takes you to the Message Object. Ok, to the `headers` field but still the Message Object 😄 That's why I did it.

**Review by fmvilas — APPROVED** [review]
👍 LGTM

**Review by jonaslagoni — APPROVED** [review]
LGTM 👍

**Review by smoya — APPROVED** [review]
LGTM!! 🚀 🌔

**Comment by derberg** [comment]
@fmvilas linked proposal is at stage 1. Is this really `docs:` like change? There is also no implementation in the Parser atm (https://github.com/asyncapi/parser-js/issues/321). This change would introduce Parser 2.0 release.

I'm all in with pushing it to the Parser and having 2.0 release, my feeling is that we have here a kind of unofficial fear of major releases, but I say I do not care that 2.0 will only have one new model, we should do it and improve UX and not look back (Jest.js had already 27 releases, React 17 and this is one of their strong points).

anyway, before we merge it, the proposal needs to reach stage 3 imho 🤷🏼

**Comment by char0n** [comment]
@derberg how can I help to push this to stage 3?

**Comment by fmvilas** [comment]
@derberg I don't think this proposal should be linked to the one on the parser. To be honest, this is just a docs reorganization. I understand they're related but IMHO parser implementation of types shouldn't be a blocker for this one. Am I missing something?

**Comment by derberg** [comment]
@fmvilas I think you are right 👍🏼

**Review by derberg — APPROVED** [review]

**Pull request merged**